packages/webapp/src/components/Modals/QuickAssignModal/index.jsx

```diff
@@ -60,24 +113,134 @@ export default function TaskQuickAssignModal({
   };
 
   const onAssign = () => {
-    assignAll && checkUnassignedTaskForSameDate() && selectedWorker.value !== null
+    const assigneeUserId = selectedWorker.value;
+
+    assignAll && checkUnassignedTaskForSameDate() && assigneeUserId !== null
       ? onAssignTasksOnDate({
           task_id: task_id,
           date: due_date,
-          assignee_user_id: selectedWorker.value,
+          assignee_user_id: assigneeUserId,
         })
       : onAssignTask({
           task_id: task_id,
-          assignee_user_id: selectedWorker.value,
+          assignee_user_id: assigneeUserId,
         });
+
+    if (isYesOptionSelected(selectedHourlyWageAction)) {
+      const wage = +parseFloat(hourlyWage).toFixed(2);
+
+      if (selectedHourlyWageAction === hourlyWageActions.SET_HOURLY_WAGE) {
+        onUpdateUserFarmWage({ user_id: assigneeUserId, wage: { type: 'hourly', amount: wage } });
+      } else if (selectedHourlyWageAction === hourlyWageActions.FOR_THIS_TASK) {
+        onChangeTaskWage(wage);
+      }
+    } else if (selectedHourlyWageAction === hourlyWageActions.DO_NOT_ASK_AGAIN) {
+      onSetUserFarmWageDoNotAskAgain({ user_id: assigneeUserId });
+    }
+
     dismissModal();
   };
 
-  const onCheckedAll = () => {
-    setAssignAll(!assignAll);
-  };
+  const disabled = !selectedWorker || !isValid;
 
-  const disabled = selectedWorker === null;
+  useEffect(() => {
+    let shouldShowWageSection = false;
+
+    if (user.is_admin && selectedWorker) {
+      const assigned = selectedWorker.label !== unAssignedOption.label;
+
+      if (assigned) {
+        const { amount } = selectedWorker.wage;
+        const hasWage = !!(amount || wageAtMoment);
+        shouldShowWageSection = !hasWage && !selectedWorker.doNotAskAgain;
+      }
+    }
+    setShowHourlyWageSection(shouldShowWageSection);
+  }, [user.is_admin, selectedWorker, wageAtMoment]);
+
+  useEffect(() => {
+    if (!user.is_admin) {
+      return;
+    }
+
+    resetField(HOURLY_WAGE_ACTION);
+    resetField(HOURLY_WAGE);
+  }, [user.is_admin, selectedWorker]);
+
+  useEffect(() => {
+    let shouldShowWageInput = false;
+
+    if (selectedHourlyWageAction) {
+      shouldShowWageInput = isYesOptionSelected(selectedHourlyWageAction);
+    }
+    if (!shouldShowWageInput) {
+      resetField(HOURLY_WAGE);
+    }
+
+    setShowHourlyWageInput(shouldShowWageInput);
+  }, [selectedHourlyWageAction]);
+
+  const radioOptions = [
+    {
+      label: t('ADD_TASK.HOURLY_WAGE.SET_HOURLY_WAGE'),
+      value: hourlyWageActions.SET_HOURLY_WAGE,
+    },
+    {
+      label: t('ADD_TASK.HOURLY_WAGE.FOR_THIS_TASK'),
+      value: hourlyWageActions.FOR_THIS_TASK,
+    },
+    {
+      label: t('common:NO'),
+      value: hourlyWageActions.NO,
+    },
+    {
+      label: t('ADD_TASK.HOURLY_WAGE.DONT_ASK'),
+      value: hourlyWageActions.DO_NOT_ASK_AGAIN,
+    },
+  ];
+
+  const renderHourlyWageSection = () => {
```

Comment:
Should create a new component or useMemo/useCallback

component overhead is bigger than the function. component will be recreated on each render